packages/requests/src/lib/requests-result.ts

```diff
@@ -26,12 +28,13 @@ export interface LoadingRequestResult extends BaseRequestResult {
   fetchStatus: 'fetching' | 'idle';
 }
 
-export interface SuccessRequestResult extends BaseRequestResult {
+export interface SuccessRequestResult<TData = any> extends BaseRequestResult {
   isLoading: false;
   isSuccess: true;
   isError: false;
   status: 'success';
   fetchStatus: 'idle';
+  data?: TData;
```

Comment:
I don't like this name because it "conflicts" with the `data` from the `join` operator.

Reply:
renamed to responseData